integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/spi/DifferentSessionFactoriesIT.java

```diff
@@ -0,0 +1,95 @@
+
+import org.hibernate.SessionFactory;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.search.integrationtest.mapper.orm.automaticindexing.AutomaticIndexingBasicIT;
+import org.hibernate.search.mapper.orm.impl.HibernateSearchContextService;
+import org.hibernate.search.mapper.orm.mapping.HibernateOrmMapping;
+import org.hibernate.search.mapper.orm.mapping.HibernateOrmSearchManager;
+import org.hibernate.search.util.SearchException;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.integrationtest.orm.OrmSetupHelper;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Tests the creation of a {@link HibernateOrmSearchManager}, using a {@link HibernateOrmMapping} with an {@link EntityManager} owned by a different {@link SessionFactory}.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class DifferentSessionFactoriesIT {
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public OrmSetupHelper ormSetupHelper = new OrmSetupHelper();
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private SessionFactory sessionFactory;
+	private SessionFactory sessionFactoryAlt;
+	private EntityManager entityManager;
+
+	@Before
+	public void setup() {
+		sessionFactory = initABasicSessionFactory();
+		sessionFactoryAlt = initABasicSessionFactory();
+		entityManager = sessionFactory.createEntityManager();
+	}
+
+	@Test
+	public void tryToUseDifferentSessionFactories() {
+		thrown.expect( SearchException.class );
+		thrown.expectMessage( "Mapping service cannot create a search manager using a different session factory." );
+
+		// service mapping is taken from the alternative session factory
+		HibernateOrmMapping serviceMapping = sessionFactoryAlt.unwrap( SessionFactoryImplementor.class )
+			.getServiceRegistry().getService( HibernateSearchContextService.class ).getMapping();
+
+		// try to use an entityManager owned by the original session factory instead
+		serviceMapping.createSearchManager( entityManager );
+	}
+
+	@After
+	public void teardown() {
+		if ( entityManager != null ) {
+			entityManager.close();
+		}
+		if ( sessionFactory != null ) {
+			sessionFactory.close();
+		}
+		if ( sessionFactoryAlt != null ) {
+			sessionFactoryAlt.close();
+		}
+	}
+
+	private SessionFactory initABasicSessionFactory() {
+		backendMock.expectSchema( AutomaticIndexingBasicIT.IndexedEntity.INDEX, b -> b
+				.field( "indexedField", String.class )
+				.field( "noReindexOnUpdateField", String.class )
+				.field( "indexedElementCollectionField", String.class )
+				.field( "noReindexOnUpdateElementCollectionField", String.class )
+		);
+
+		SessionFactory factory = ormSetupHelper.withBackendMock( backendMock )
+				.setup( AutomaticIndexingBasicIT.IndexedEntity.class );
```

Comment:
Do not reuse an entity from another test, create your own inside the test class. It can be as simple as this:

```
	@Entity(name = "indexed")
	@Indexed(name = IndexedEntity.INDEX)
	public static class IndexedEntity {
		public static String INDEX = "IndexedEntity";

		@Id
		Integer id;

		@Id
		public Integer getId() {
			return id;
		}

		public void setId(Integer id) {
			this.id = id;
		}
	}
```

... but please, don't reuse entity classes across tests. I don't want changing one class in one test to affect other tests...

Reply:
I agree with you, I also prefer decoupling over reuse.